test/BasicToken.js

```diff
@@ -1,49 +1,32 @@
 contract('BasicToken', function(accounts) {
 
-  it("should return the correct totalSupply after construction", function(done) {
-    return BasicTokenMock.new(accounts[0], 100)
-      .then(function(token) {
-        return token.totalSupply();
-      })
-      .then(function(totalSupply) {
-        assert.equal(totalSupply, 100);
-      })
-      .then(done);
+  it("should return the correct totalSupply after construction", async function(done) {
+    let token = await BasicTokenMock.new(accounts[0], 100);
+    let totalSupply = await token.totalSupply();
+    assert.equal(totalSupply, 100);
+    done();
```

Comment:
I have 2 or 3 tests that require the done. I basically went with consistency. Ill remove the done from all the other tests.